src/components/app.tsx

```diff
@@ -10,11 +10,21 @@ import {VoyagerConfig} from '../models/config';
 import {State} from '../models/index';
 import {AppRoot} from './app-root';
 
+export interface VoyagerData {
+  data: Data;
+  filename: string;
+}
+
+export interface SpecWithFilename {
+  spec: TopLevel<FacetedCompositeUnitSpec>;
+  filename: string;
+}
+
 export interface Props extends React.Props<App> {
   config?: VoyagerConfig;
-  data?: Data;
+  data?: VoyagerData;
```

Comment:
Instead of making two properties having "filename", why don't we make filename a new prop of the application?